### PR TITLE
Upgrade to Consul 1.0.0

### PR DESCRIPTION
**Pull request by jwreagor**
This commit bumps the blueprint to Consul 1.0.0 REL. A few things have changed in our Consul cluster pattern that are worth noting.

* HCL is now Consul's configuration format
* `recursor` has been renamed to a list called `recursors`
* Remove Consul UI downloading as an extra step in Dockerfile
* Consul UI is now included in the dist so we've enabled it by default
* Bump Alpine to 3.6 as our base images
* Bump ContainerPilot to 3.6.0

Fixes #42 

/cc @misterbisson @geek @tianon

### CODE REVIEW

**Comment by jwreagor** [comment]
~~This doesn't address testing. While the tests passed for me, I had a blocker with one of them and want to circle back in full at a later date.~~ Resolved!

File: test/compose.sh [hunk 1]
```diff
@@ -277,4 +277,4 @@ test-quorum-consistency() {
 test-rejoin-raft 3
 test-rejoin-raft 5
 test-graceful-leave
-test-quorum-consistency
+#test-quorum-consistency
```
Comment (cjihrig) [inline test/compose.sh]: If you're going to leave this in, but commented out, can you add to the comment and explain why.

**Review by cjihrig — APPROVED** [review]

File: test/compose.sh (hunk 1 above)
Comment (jwreagor) [reply test/compose.sh]: You're right, I haven't seen it work even on the previous versions but that probably deserves more of an explanation.

**Comment by tianon** [comment]
This all looks generally sane to me! :+1:

File: test/compose.sh (hunk 1 above)
Comment (jwreagor) [reply test/compose.sh]: Spent more than 2 minutes looking at the problem and it appears the output changed between versions. Patch incoming. 👍

**Pull request merged**